troubleshooting.md

```diff
@@ -0,0 +1,333 @@
+1. Walk through the list of [image build object types](#system-object-types) and check if any objects are missing. For any missing objects, check the logs of the [TAP](#tap-logs) or [Application Service Adapter](#application-logs) component listed in the "Created By" column to see if there are any error messages.
+1. If the component listed in the "Created By" column does not have any helpful error messages, identify the previous object in the image build chain and check the logs of the component listed in the "Reconciled By" column to see if there are any error messages.
+
+### Pushing an app fails to upload an image to the image registry
+#### Symptom
+When I run `cf push`, the following output/error message is returned:
+```bash
+    Unexpected Response
+    Response Code: 500
+    Code: 0, Title: , Detail: {"errors":[{"detail":"An unknown error occurred.","title":"UnknownError","code":10001}]}
+```
+#### Possible Causes
+The image registry is unavailable/unauthorized.
+#### Troubleshooting Steps/Potential Solutions
+1. Check the `korifi-api-deployment` [logs](#application-logs) for registry related failures like `failed to upload image`.
+1. Determine the cause of the upload failure like connection refused, credentials failed, certificate untrusted, etc.
+1. Check the image registry settings provided to the Application Service Adapter installation related to the failure and reinstall Application Service Adapter with the correct image registry settings.
+
+### Pushing an app fails to start
+#### Symptom
+When I run `cf push`, my app stages correctly, but fails to start or become healthy.
+#### Possible Causes
+1. There is an issue with the application's code.
+1. The Kubernetes cluster cannot schedule the application.
+1. The Application Service Adapter cannot turn the app into a StatefulSet.
+#### Troubleshooting Steps/Potential Solutions
+1. To debug the application's code, check the [application logs](#cfapp-logs)
+1. To debug applications that fail to be scheduled, check the [Kubernetes System Logs](#system-events) and the [StatefulSet](#system-object-types) status in the CFSpace namespace.
+1. Walk through the list of [run app object types](#system-object-types) and check if any objects are missing. For any missing objects, check the logs of the [TAP](#tap-logs) or [Application Service Adapter](#application-logs) component listed in the "Created By" column to see if there are any error messages.
+1. If the component listed in the "Created By" column does not have any helpful error messages, identify the previous object in the run app chain and check the logs of the component listed in the "Reconciled By" column to see if there are any error messages.
+
+### Deployed apps fail to become routable
+#### Symptom
+When I run `cf push`, my app stages and runs, but fails to be routable.
+#### Possible Causes
+There is an issue with the Application Service Adapter Deployments
```

Comment:
I think we should also mention that they should look at the `HTTPProxy` and `CFRoute` resources that correspond with the application's route. The `HTTPProxy` usually has pretty detailed conditions in its status.

Reply:
Done. Added HTTPProxy to the objects table